src/printer.js

```diff
@@ -609,24 +605,45 @@ function genericPrintNoParens(path, options, print) {
     case "ReturnStatement":
       parts.push("return");
 
-      if (
-        n.argument &&
-        n.argument.comments &&
-        n.argument.comments.some(comment => comment.leading)
-      ) {
-        parts.push(
-          concat([
-            " (",
-            indent(
-              options.tabWidth,
-              concat([softline, path.call(print, "argument")])
-            ),
-            line,
-            ")"
-          ])
-        );
-      } else if (n.argument) {
-        parts.push(" ", path.call(print, "argument"));
+      if (n.argument) {
+        const comments = n.argument.comments || [];
+        if (
+          n.argument.type === "BinaryExpression" ||
+          n.argument.type === "LogicalExpression" ||
+          n.argument.type === "ConditionalExpression" ||
+          n.argument.type === "CallExpression" ||
+          n.argument.type === "MemberExpression" ||
+          n.argument.type === "SequenceExpression" ||
+          n.argument.type === "TaggedTemplateExpression"
+        ) {
+          function getLeftSide(node) {
```

Comment:
for example, if someone threw an AssignmentExpression in there as the "object" of a MemberExpression (`(a=b).b`), it would traverse the AssignmentExpression too, since it has a `left` attribute. 

I don't think that's a problem, but figured it should be mentioned